studio/components/interfaces/TableGridEditor/TableGridEditor.tsx

```diff
@@ -277,26 +236,21 @@ const TableGridEditor = ({
   }
 
   const onSelectEditColumn = async (name: string) => {
-    // For some reason, selectedTable here is stale after adding a table
-    // temporary workaround is to list grab the selected table again
-    const tables: PostgresTable[] = meta.tables.list()
-    // @ts-ignore
-    const table = tables.find((table) => table.id === Number(tableId))
-    const column = find(table!.columns, { name }) as PostgresColumn
+    const column = find(selectedTable?.columns ?? [], { name }) as PostgresColumn
     if (column) {
-      onEditColumn(column)
+      snap.onEditColumn(column)
     } else {
-      console.error(`Unable to find column ${name} in ${table?.name}`)
+      console.error(`Unable to find column ${name} in ${selectedTable?.name}`)
     }
   }
 
   const onSelectDeleteColumn = async (name: string) => {
-    // For some reason, selectedTable here is stale after adding a table
-    // temporary workaround is to list grab the selected table again
-    const tables: PostgresTable[] = meta.tables.list()
-    const table = tables.find((table) => table.id === Number(tableId))
-    const column = find(table!.columns, { name }) as PostgresColumn
-    onDeleteColumn(column)
+    const column = find(selectedTable?.columns ?? [], { name }) as PostgresColumn
+    if (column) {
+      snap.onDeleteColumn(column)
+    } else {
+      console.error(`Unable to find column ${name} in ${selectedTable?.name}`)
```

Comment:
Can we make this into a toast?
```ui.setNotification({ category: 'error', message: '...', error: <some error> })```

technically we shouldn't run into this block, but if for some reason we do, at least the issue is surfaced to the user